pkg/discovery/servicehandler.go

```diff
@@ -110,6 +99,26 @@ func (s *serviceHandler) getServiceDetail(asset *anypoint.Asset, api *anypoint.A
 	}
 	authPolicy, configuration, isSlaBased := getAuthPolicy(policies)
 
+	// TODO can be refactored to not use authPolicy in checksum and use policy
+	isAlreadyPublished, checksum := isPublished(api, authPolicy, s.cache)
+	// If true, then the api is published and there were no changes detected
+	if isAlreadyPublished {
+		logger.WithFields(logrus.Fields{"policy": authPolicy, "msg": "api is already published"})
+		return nil, nil
+	}
+
+	// TODO Handle purging of cache
+	// Set the item so that it can be found by the agent-sdk for validation
+	cacheKey := common.FormatAPICacheKey(fmt.Sprint(asset.ID), api.ProductVersion)
```

Comment:
Not sure we really need this since the sdk tries to validate based on the API Service because our revisions do not reflect the API Service that actually exists.